pypesto/visualize/misc.py

```diff
@@ -309,3 +285,96 @@ def apparent_composite_color_component(
         apparent_composite_color_component(fg[i], bg[i])
         for i in range(LEN_RGB)
     ]
+
+
+def process_start_indices(
+    result: Result,
+    start_indices: Union[str, int, Iterable[int]] = ALL,
+) -> np.ndarray:
+    """
+    Process the start_indices.
+
+    Create an array of indices if a number was provided and checks that the
+    indices do not exceed the max_index.
+
+    Parameters
+    ----------
+    start_indices:
+        list of indices or int specifying an endpoint of the sequence of
+        indices. Furthermore the following strings are possible:
+            * 'all', this is the default, using all start indices.
+            * 'all_clustered', this includes the best  start and all that are
+            in a cluster of size > 1.
+            * 'first_cluster', includes all starts that belong to the first
+            cluster.
+    result:
+        Result to determine maximum allowed length and/or clusters.
+    """
+    if isinstance(start_indices, str):
+        if start_indices == ALL:
+            return np.asarray(range(len(result.optimize_result)))
+        elif start_indices == ALL_CLUSTERED:
+            clust_ind, clust_size = assign_clusters(
+                delete_nan_inf(result.optimize_result.fval)[1]
+            )
+            # get all clusters that have size > 2:
+            clust_gr2 = np.where(clust_size > 2)[0]
```

Comment:
and what if the best value occurs only once? will the best start still be included as the docsring says?